DSharpPlus/Net/Rest/DiscordApiClient.cs

```diff
@@ -1004,7 +1003,8 @@ internal async Task<DiscordMessage> EditMessageAsync(ulong channel_id, ulong mes
                 HasContent = content.HasValue,
                 Content = content.HasValue ? (string)content : null,
                 HasEmbed = embed.HasValue,
-                Embed = embed.HasValue ? (DiscordEmbed)embed : null
+                Embed = embed.HasValue ? (DiscordEmbed)embed : null,
+                Components = components.HasValue ? components.Value.ToList().AsReadOnly() : null
```

Comment:
`ToList().AsReadOnly()` is completely unnecessary here, it's already the right type.